api/src/main/java/org/openmrs/api/db/hibernate/HibernatePatientDAO.java

```diff
@@ -435,19 +435,42 @@ public PatientIdentifierType getPatientIdentifierType(Integer patientIdentifierT
 	private String getDuplicatePatientsSQLString(List<String> attributes) {
 		StringBuilder outerSelect = new StringBuilder("select distinct t1.patient_id from patient t1 ");
 		final String t5 = " = t5.";
-		
+
+		List<String> sortedAttributes = new ArrayList<>();
+		// sort attributes: patient field names go first, person field names go second etc.
```

Comment:
Isn't this a brittle implementation? Where sql query passes or fails depending on order of parameters?

Reply:
I agree it's not the best implementation but I didn't want to mess with the legacy code for now. The solution above fixes the method. However, we could create another ticket to fully refactor it.

Reply:
Why not use this ticket?

Reply:
I'm just not sure how much time it's gonna take, but I could try to do it here :)

Reply:
Give it a shot and see how it goes. If it turns out to take long, we can then leave it.

Reply:
Got it